codex-rs/core/src/codex.rs

```diff
@@ -1873,6 +1879,9 @@ async fn run_compact_task(
     }
 
     sess.remove_task(&sub_id);
+
+    sess.trim_history_to_last_messages(1);
```

Comment:
Should we just inline this since it is only called once?

```
{
    let mut state = sess.state.lock_unchecked();
    state.history.keep_last_messages(1);
}
```